octobot_evaluators/evaluator/strategy_evaluator.py

```diff
@@ -57,6 +135,244 @@ async def matrix_callback(self,
+        ).keys()
+
+    def get_available_time_frames(self,
+                                  matrix_id,
+                                  exchange_name=None,
+                                  tentacle_type=None,
+                                  cryptocurrency=None,
+                                  symbol=None,
+                                  use_cache=True):
+        if use_cache:
+            try:
+                return self.available_time_frames_cache[matrix_id][exchange_name][tentacle_type][cryptocurrency][symbol]
+            except KeyError:
+                available_time_frames = StrategyEvaluator._get_available_time_frames(
+                    matrix_id, exchange_name,tentacle_type, cryptocurrency, symbol)
+                if matrix_id not in self.available_time_frames_cache:
+                    self.available_time_frames_cache[matrix_id] = {}
+                if exchange_name not in self.available_time_frames_cache[matrix_id]:
+                    self.available_time_frames_cache[matrix_id][exchange_name] = {}
+                if tentacle_type not in self.available_time_frames_cache[matrix_id][exchange_name]:
+                    self.available_time_frames_cache[matrix_id][exchange_name][tentacle_type] = {}
+                if cryptocurrency not in self.available_time_frames_cache[matrix_id][exchange_name][tentacle_type]:
+                    self.available_time_frames_cache[matrix_id][exchange_name][tentacle_type][cryptocurrency] = {
+                        symbol: available_time_frames
+                    }
+                return available_time_frames
+        return StrategyEvaluator._get_available_time_frames(matrix_id, exchange_name,
+                                                            tentacle_type, cryptocurrency, symbol)
+
+    @staticmethod
+    def _get_available_time_frames(matrix_id,
+                                   exchange_name,
+                                   tentacle_type,
+                                   cryptocurrency,
+                                   symbol):
+        try:
+            evaluator_nodes = get_node_children_by_names_at_path(matrix_id,
+                                                                 get_tentacle_path(exchange_name=exchange_name,
+                                                                                   tentacle_type=tentacle_type))
+            first_node = next(iter(evaluator_nodes.values()))
+            return get_node_children_by_names_at_path(matrix_id,
+                                                      get_tentacle_value_path(cryptocurrency=cryptocurrency,
+                                                                              symbol=symbol),
+                                                      starting_node=first_node).keys()
+        except StopIteration:
+            return []
+
+    def get_available_symbols(self, matrix_id, exchange_name,
```

Comment:
Where is this method used ?

Reply:
In strategies implementations, ex:
![image](https://user-images.githubusercontent.com/9078616/80135045-ed8daa00-859f-11ea-82f1-bd4c53f525ed.png)

Reply:
ok 👍